test/dynamo/test_misc.py

```diff
@@ -6533,6 +6533,26 @@ def deep(c):
         self.assertEqual(eager, compiled)
         self.assertEqual(counter.frame_count, 1)
 
+    def test_deque_input(self):
+        a = torch.randn([2, 3])
+        b = torch.randn([2, 3])
+        d1 = collections.deque([a, b])
+        d1.insert(0, "foo")
+
+        d2 = collections.deque([a, b])
+        d2.insert(0, "foo")
+
+        def fn(q):
+            a = q.pop()
+            b = q.pop()
```

Comment:
This doesn't seem like it exercises all of the methods you added

Reply:
added more tet